pkg/cache/v3/linear.go

```diff
@@ -144,6 +160,16 @@ func WithLogger(log log.Logger) LinearCacheOption {
 	}
 }
 
+// WithSotwStableVersions changes the versions returned in sotw to encode the list of resources known
+// in the subscription.
+// The use of stable versions for sotw also deduplicates updates to clients if the cache updates are
+// not changing the content of the resource.
+func WithSotwStableVersions() LinearCacheOption {
```

Comment:
Should we even make this something the client can set? Or if we do can we explain in what circumstances you would want to set it or not set it?

 If I understand correctly the drawback of the SotwStableVersions is that you have to sort and marshal every request that comes in so it's less CPU efficient, but clients don't get unnecessary updates?
 
It would be nice if we could settle on one way of doing things. Do we need additional loadtesting to understand why to decide to pick one version over the other, or can we just drop the non-stable-version STOW?

Reply:
This is an opt-in flag as this greatly changes the behavior by: 
 - no longer resending resources every time they are added to the cache, even if identical
 - use more CPU to compute the resource versions
 - the versions returned are no longer a monotonic count users might be relying on. I might remove this later on, but I did not want to mandate this for now